### PR TITLE
Create page of the register decimate

### LINKED ISSUE #9: Create route page for register decimate
- [x] Create integration with API fake;
- [x] Use NgRX to state management;
- [x] Form create for save members.

### PR DESCRIPTION
**Pull request by josileudo**
### Description

- Create a dropdown menu with routes;
- Create a form to save decimate;
- Config `routes` file;

### Have a style change?

Kapture 2023-08-05 at 16 51 31

#### Please check if your PR fulfills the following requirements:

Add a new route to register a decimate.

#### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

Closes #9 

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/app/pages/Register/index.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+  constructor(label = 'Save Decimate', loading = false) {
+    this.label = label;
+    this.loading = loading;
+  }
+}
+@Component({
+  selector: 'page-register',
+  standalone: true,
+  imports: [
+    RouterModule,
```
Comment (luiz258) [inline src/app/pages/Register/index.ts:54]: I think I could create a Shared Module to import these basic Modules, I'm seeing several repeats and the code is getting long

File: src/main.ts [hunk 2]
```diff
@@ -6,19 +6,36 @@ import {
...
-    provideStore({ theme: themeReducer }),
+    importProvidersFrom(
+      BrowserModule,
+      BrowserAnimationsModule,
+      HttpClientModule
+    ),
+    provideStore(),
+    provideState(createMemberFeature),
+    provideState(messageFeature),
+    provideState(themeFeature),
```
Comment (luiz258) [inline src/main.ts:35]: so podemos declarar so um estado dentro do provideState?

provideState(createMemberFeature, messageFeature, themeFeature),
É viavel fazer dessa forma acima?

File: src/styles.scss [hunk 3]
```diff
@@ -26,6 +26,21 @@ body {
...
+button,
+a {
+  cursor: pointer;
+}
+
+a {
+  text-decoration: none;
+  color: var(--text-color);
+
+  // transition: all 0.2s;
```
Comment (luiz258) [inline src/styles.scss:38]: remove comment

**Review by luiz258 — CHANGES_REQUESTED** [review]

File: src/app/pages/Register/index.ts (hunk 1 above)
Comment (josileudo) [reply src/app/pages/Register/index.ts:54]: This is a big suggestion, but this will escape the standalone architecture

File: src/main.ts (hunk 2 above)
Comment (josileudo) [reply src/main.ts:35]: No, this way
 ```js
provideState(createMemberFeature, messageFeature, themeFeature)
```
 I can create a file, for example, a index.ts, and called a function and input this 
 
 ```js
 export function provideRegisterFeature(): EnvironmentProviders {
  return makeEnvironmentProviders([
    provideState(createMemberFeature),
    provideState(messageFeature),
    provideState(themeFeature),
  ]);
}
 ```
 
 So, importing here 
 ```js
 
bootstrapApplication(AppComponent, {
  providers: [
    importProvidersFrom(
      BrowserModule,
      BrowserAnimationsModule,
      HttpClientModule
    ),
    provideStore(),
    provideRegisterFeature(),
    provideRouter(routes),
    provideStoreDevtools(),
    provideEffects(CreateMemberEffect),
    { provide: LOCALE_ID, useValue: 'pt' },
    { provide: DEFAULT_CURRENCY_CODE, useValue: 'BRL' }
  ]
});
 ```

File: src/styles.scss (hunk 3 above)
Comment (josileudo) [reply src/styles.scss]: fixed

**Pull request merged**